drivers/watchdog/wdt_ti_rti.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2025 Ayush Singh, BeagleBoard.org Foundation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_j7_rti_wdt
+
+#include <zephyr/drivers/watchdog.h>
+#include <zephyr/kernel.h>
+#include <stdint.h>
+
+#define WDENABLE_KEY  0xa98559da
+#define WDDISABLE_KEY 0x5312aced
+
+#define WDKEY_SEQ0 0xe51a
+#define WDKEY_SEQ1 0xa35c
+
+#define WDT_PRELOAD_SHIFT 13
+
+#define WDT_PRELOAD_MAX 0xfff
+
+#define RTIWWDRX_NMI   0xa
+#define RTIWWDRX_RESET 0x5
+
+#define RTIGCTRL_HALT_BY_DBG 0
+#define RTIGCTRL_RUN_BY_DBG  BIT(15)
+
+#define RTIWWDSIZE_100P  0x5
+#define RTIWWDSIZE_50P   0x50
+#define RTIWWDSIZE_25P   0x500
+#define RTIWWDSIZE_12P5  0x5000
+#define RTIWWDSIZE_6P25  0x50000
+#define RTIWWDSIZE_3P125 0x500000
+
+#define DEV_CFG(dev)  ((const struct wdt_ti_rti_config *)(dev)->config)
+#define DEV_DATA(dev) ((struct wdt_ti_rti_data *)(dev)->data)
+#define DEV_REGS(dev) ((struct wdt_ti_rti_regs *)DEVICE_MMIO_GET(dev))
+
+struct wdt_ti_rti_regs {
+	volatile uint32_t GCTRL;
+	uint32_t pad[35];
+	volatile uint32_t DWDCTRL;
+	volatile uint32_t DWDPRLD;
+	volatile uint32_t WDSTATUS;
+	volatile uint32_t WDKEY;
+	volatile uint32_t DWDCNTR;
+	volatile uint32_t WWDRXNCTRL;
+	volatile uint32_t WWDSIZECTRL;
+};
```

Comment:
Might be useful to add a very brief description and offset next to the fields, For example:

```c
	volatile uint32_t DWDCTRL;  /**< Digital Watchdog Control Register, offset: 0x90 */
```

Reply:
Added